### PR TITLE
Added an example page for setting up custom-auth-flows

### PR DESCRIPTION
**Pull request by yuhengshs**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/functions/examples/custom-auth-flows/index.mdx [hunk 1]
```diff
@@ -0,0 +1,215 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Custom Auth Challenge',
+  description:
+    'Leverage Custom Auth with and without SRP, Allowing for a series of challenge and response cycles that can be customized to meet different requirements during sign in.',
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/functions/examples/custom-auth-flows/index.mdx:6]:
```suggestion
    'Leverage Custom Auth with and without SRP, allowing for a series of challenge and response cycles that can be customized to meet different requirements during sign in.',
```

File: src/pages/[platform]/build-a-backend/functions/examples/custom-auth-flows/index.mdx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+```
+
+Or if you are using `CUSTOM_WITH_SRP`:
+
+```ts title="amplify/auth/define-auth-challenge/handler.ts"
+import type { DefineAuthChallengeTriggerHandler } from "aws-lambda"
+
+export const handler: DefineAuthChallengeTriggerHandler = async (event) => {
+  if (event.request.session.length === 0) {
+    event.response.issueTokens = false;
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/functions/examples/custom-auth-flows/index.mdx:128]: for folks seeing this trigger for the first time it's tough to grok what each of these steps mean or are doing, can we add a short comment like "verify password challenge" inside each of the conditional blocks?

File: src/pages/[platform]/build-a-backend/functions/examples/custom-auth-flows/index.mdx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  },
+  triggers: {
+    createAuthChallenge,
+    defineAuthChallenge,
+    verifyAuthChallengeResponse,
+  },
+})
+```
+
+After deploying the changes, whenever a user attempts to sign in with `CUSTOM_WITH_SRP` or `CUSTOM_WITHOUT_SRP`, the lambda challenges will be triggered.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/functions/examples/custom-auth-flows/index.mdx:215]:
```suggestion
After deploying the changes, whenever a user attempts to sign in with `CUSTOM_WITH_SRP` or `CUSTOM_WITHOUT_SRP`, the Lambda challenges will be triggered.
```

**Review by josefaidt — CHANGES_REQUESTED** [review]
Hey @yuhengshs I like these examples, thank you for taking the time to file this! I left a few small comments, but can we also add a callout/link to this page from https://docs.amplify.aws/react/build-a-backend/auth/connect-your-frontend/switching-authentication-flows/#custom_with_srp--custom_without_srp-flows ?

File: src/pages/[platform]/build-a-backend/auth/connect-your-frontend/switching-authentication-flows/index.mdx [hunk 4]
```diff
@@ -189,6 +189,8 @@ if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE') {
   await confirmSignIn({ challengeResponse });
 }
 ```
+To create a CUSTOM_AUTH challenge with a Lambda Trigger, please visit [AWS Amplify Custom Auth Challenge example](/[platform]/build-a-backend/functions/examples/custom-auth-flows/) for set up instructions. 
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/connect-your-frontend/switching-authentication-flows/index.mdx:193]:
```suggestion

To create a CUSTOM_AUTH challenge with a Lambda Trigger, please visit [AWS Amplify Custom Auth Challenge example](/[platform]/build-a-backend/functions/examples/custom-auth-flows/) for setup instructions. 
```

**Review by josefaidt — APPROVED** [review]

**Review by josefaidt — APPROVED** [review]

**Review by cwomack — APPROVED** [review]

**Review by jacoblogan — APPROVED** [review]

**Review by aspittel — APPROVED** [review]

**Review by josefaidt — APPROVED** [review]

**Pull request merged**